File: src/main/webapp/resources/angularjs/services/exportService.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


var app = angular.module("rhApp");

app.constant("exportType", "export/exportFile"),
app.constant("readExcel", "export/readExcel").factory("exportService", function ($http, exportType,readExcel) {

    return{
        exportType: function (type, columns) {
            var ajaxget = $http({
                url: exportType + "/" + type + "/" + columns,
                method: 'GET',
                async: true,
                cache: false,
                headers: {'Accept': 'application/json', 'Pragma': 'no-cache'}
            });
           
        },
        readExcel: function () {
            var ajaxget = $http({
                url: readExcel,
                method: 'GET',
                async: true,
                cache: false,
                headers: {'Accept': 'application/json', 'Pragma': 'no-cache'}
            });
           
        }
    };

});

File: src/main/java/br/sp/telesul/service/ExportService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package br.sp.telesul.service;

import br.sp.telesul.model.Funcionario;
import java.util.List;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.springframework.stereotype.Service;

/**
 *
 * @author ebranco
 */
@Service
public interface ExportService {

    public void buildExcelDocument(String type, List<String> columns, List<String> columnsFormacao, List<String> columnsIdiomas, List<String> columnsCertificacoes, HttpServletRequest request, HttpServletResponse response);
    public List<Funcionario> readExcelDocument();

}
